examples/custom-tool-and-agent/README.md

```diff
@@ -0,0 +1,257 @@
+     
+      ```bash
+      python --version
+      ```
+
+2. **[Google Cloud CLI](https://cloud.google.com/sdk/docs/install)**:
+   - The `gcloud` CLI is required for authenticating to access private package repositories. Ensure it's installed and configured (`gcloud auth login`, `gcloud config set project <your-project-id>`).
+   - Install the CLI by following the instructions at [cloud.google.com/sdk/docs/install](https://cloud.google.com/sdk/docs/install).
+   - After installation, configure it by running:
+     
+       ```bash
+       gcloud auth login
+       gcloud config set project <your-project-id>
+       ```
+
+3. **VSCode IDE**
+   - Go to [VSCode](https://code.visualstudio.com/download) to download VSCode IDE.
+
+4. **GDP Labs VPN**
+   - The GDP Labs VPN is required to access the GLChat staging environment where tools are uploaded and agents are tested. Ensure you are connected to the VPN before proceeding with steps involving GLChat.
+
+### Installing dependencies
+
+1. Clone the `gen-ai-examples` repository.
+
+   ```bash
+   git clone https://github.com/GDP-ADMIN/gen-ai-examples.git
+   ```
+
+2. Navigate to the example directory:
+
+   ```bash
+   cd gen-ai-examples/examples/custom-tool-and-agent
+   ```
+
+3. Execute the script:
+
+   - For Linux, macOS, or Windows WSL:
+
+     ```bash
+     ./local-start.sh
+     ```
+
+     > [!NOTE]
+     > On Windows, you can either install [WSL (Windows Subsystem for Linux)](https://learn.microsoft.com/en-us/windows/wsl/install) or execute the batch file in Windows Powershell or Command Prompt as described in the next section.
+
+   - For Windows Powershell:
+
+     ```powershell
+     .\local-start.bat
+     ```
+
+   - For Windows Command Prompt:
+
+     ```cmd
+     local-start.bat
+     ```
+
+## Tool Development Guide
+
+### Creating a Custom Tool
+1. Open your VSCode within the `gen-ai-examples/examples/custom-tool-and-agent` directory.
+2. Activate the newly installed virtual environment. VSCode usually activates this environment automatically. If it doesn't activate the environment automatically, then you should click it and select the python from this path `/custom-tool-and-agent/.venv/bin/python`.
+   *   **Alternative**: You can also use the Command Palette (`Ctrl+Shift+P` or `Cmd+Shift+P`), type "Python: Select Interpreter", and choose the interpreter associated with the `.venv` directory within your project.
+   *   **Verify**: Check the bottom status bar in VSCode (as shown in the image below) to ensure the selected Python interpreter points to the `.venv` environment created by Poetry (e.g., `Python 3.11.10 ('.venv')`).
+
+         <img width="203" alt="image" src="https://github.com/user-attachments/assets/3dffba0d-8cd7-4577-880b-ea74d0080b7c" />
+
+3. Create a new Python file named `weather_forecast_tool.py` in the current directory (`examples/custom-tool-and-agent`).
```

Comment:
I think we don't need to ask user to create a file(?) Let's just use the existing `weather_forecast_tool.py` to be uploaded via admin dashboard. We can add description that user can change the file if they want (and then re-upload).